src/Infrastructure/Repository/TaskRepository.php

```diff
@@ -124,7 +138,42 @@ public function findById(string $taskId, array $tags = []): ?TaskInterface
         if ($tags) {
             $criteria['tags'] = $tags;
         }
+        /** @var \SprykerSdk\Sdk\Infrastructure\Entity\Task|null $task */
+        $task = $this->findOneBy($criteria);
```

Comment:
Somehow this line seems broken to me because tags are stored in the DB as a serialized string and this criteria should check the intersection between the given array and serialized array in the DB.